### PR TITLE
QueryEditorRow: Fixes issue loading query editor when data source variable selected

### LINKED ISSUE #61710: Frontend: Using a datasource variable does not display a query editor
When creating a datasource variable and attempting to use said variable a valid query editor is not displayed. 

It seems that this bug was introduced by the changes made in #60218 (verified by using a dashboard with a datasource variable in the prior commit).

### PR DESCRIPTION
**Pull request by torkelo**
Fixes issue showing query editor (not shown) after selecting data source variable as
the data source.

Introduced by https://github.com/grafana/grafana/pull/60218

Fixes: #61710

### CODE REVIEW

**Review by axelavargas — COMMENTED** [review]
Hey Torkel 👋🏾 ,  I gave it a spin and found a minor issue:

When the panel has selected a data source variable that returns an error, the error does not go away if I switch to a different data source.

https://user-images.githubusercontent.com/239999/214251616-d0b82d9c-0aac-4e08-bb9f-cf84f49a68b6.mp4

**Review by mckn — APPROVED** [review]
LGTM! The issues I fixed works as expected with this change. Thanks for having a look at this @torkelo 🙏🏻

**Comment by torkelo** [comment]
@axelavargas it is not relaterad to this, but another issue. We should clear the current data state when switching data source

**Review by axelavargas — APPROVED** [review]

**Review by ivanortegaalba — APPROVED** [review]

**Comment by marefr** [comment]
I think backporting this to 9.3.x should be enough, no?

**Comment by torkelo** [comment]
@marefr @mckn I have to remove that new wait check to get Explore tests to pass, they only pass uid via URL and not type. 

I don't think that wait clause is needed

**Comment by torkelo** [comment]
> I think backporting this to 9.3.x should be enough, no?

@marefr Why would we leave the bug in 9.1.x, 9.2.x ? (still unsure why the PR that introduced this was backported to so many branches)

File: public/app/features/query/components/QueryEditorRow.tsx [hunk 1]
```diff
@@ -238,20 +238,10 @@ export class QueryEditorRow<TQuery extends DataQuery> extends PureComponent<Prop
     }
   }
 
-  waitingForDatasourceToLoad = (): boolean => {
-    // if we not yet have loaded the datasource in state the
-    // ds in props and the ds in state will have different values.
-    return this.props.dataSource.uid !== this.state.datasource?.uid;
-  };
```
Comment (torkelo) [inline public/app/features/query/components/QueryEditorRow.tsx:245]: The log updateQueries cleans the query models so this should not be needed 
To me it looks like a bug in Alerting when handling change data source in that it is not clearing the query model correctly.

Related to
https://github.com/grafana/support-escalations/issues/4577

Comment (torkelo) [reply public/app/features/query/components/QueryEditorRow.tsx]: added it back but with a better check to make ExplorePage tests work

**Comment by marefr** [comment]
> > I think backporting this to 9.3.x should be enough, no?
> 
> @marefr Why would we leave this critical bug in 9.1.x, 9.2.x ? (still unsure why the PR that introduced this was backported to so many branches)

Sorry didn't understand this is a problem in those older releases :cry:

**Comment by kylebrandt** [comment]
This (commit 9f7ddf1f0b5234a1fd82e1f92d1444f81a794e45) seems to have made it so adding an expression in a dashboard results in an empty row (header, but not contents), will make issue.

**Comment by jiaojialin** [comment]
Hello, I wonder when this fix will be released ?   This bug has prevented us to upgrade to v.9.3.6 to use some other newer features

**Comment by gelicia** [comment]
@jiaojialin the bug was introduced in 9.3.2 and the fix should have been released in 9.3.6 along with being backported to other 9.* versions - is that not what you're experiencing?

**Comment by jiaojialin** [comment]
> @jiaojialin the bug was introduced in 9.3.2 and the fix should have been released in 9.3.6 along with being backported to other 9.* versions - is that not what you're experiencing?

Thanks for your reply!   I am sorry, I think I was mistaken and commented on the wrong PR :).  I meant to ask when another bug fix will be release https://github.com/grafana/grafana/pull/62510

**Pull request merged**